src/dymoprint/dymo_print_engines.py

```diff
@@ -119,6 +119,96 @@ def render_barcode(
+
+        Args:
+            barcode_input_text (str): The input text to be encoded in the barcode.
+            bar_code_type (str): The type of barcode to be rendered.
+
+        Returns:
+            Image: A barcode image.
+        """
+        label_height = DymoLabeler.max_bytes_per_line(self.tape_size_mm) * 8
+        if len(barcode_input_text) == 0:
+            return Image.new("1", (1, label_height))
+
+        code = barcode_module.get(
+            bar_code_type, barcode_input_text, writer=BarcodeImageWriter()
+        )
+
+        code_bitmap = code.render(
+            {
+                "font_size": 10,
+                "vertical_margin": 2,
+                "module_height": (DymoLabeler.max_bytes_per_line(self.tape_size_mm) * 6)
+                - 16,
+                "module_width": 1,
+                "background": "black",
+                "foreground": "white",
+            }
+        )
+        return code_bitmap
+
+    def render_barcode_with_text(
+        self,
+        barcode_input_text,
+        bar_code_type,
+        font_file_name: str,
+        frame_width,
+        font_size_ratio=0.9,
+        align="center",
+    ):
+        """
+        Renders a barcode image with the text below it.
+
+        Args:
+            barcode_input_text (str): The input text to be encoded in the barcode.
+            bar_code_type (str): The type of barcode to be rendered.
+            font_file_name (str): The name of the font file to be used.
+            frame_width (int): The width of the frame around the text.
+            font_size_ratio (float): The ratio of font size to line height. Default
+                is 1.
+
+        Returns:
+            Image: A barcode with text image.
+        """
+        assert align in ("left", "center", "right")
+        code_bitmap = self.__create_custom_code_bitmap(
+            barcode_input_text, bar_code_type
+        )
+
+        text_bitmap = self.render_text(
+            barcode_input_text, font_file_name, frame_width, font_size_ratio, align
+        )
+        text_width = int(
+            text_bitmap.width // (text_bitmap.height / (code_bitmap.height / 2))
+        )
+        text_bitmap = text_bitmap.resize((text_width, code_bitmap.height // 2))
+
+        # Merge two images into one bitmap
+        code_with_text_bitmap = Image.new(
+            "RGB", (code_bitmap.width, 3 * code_bitmap.height // 2), "black"
+        )
+        code_with_text_bitmap.paste(code_bitmap, (0, 0))
+        match align:
+            case "left":
+                code_with_text_bitmap.paste(text_bitmap, (0, code_bitmap.height))
+            case "center":
+                code_with_text_bitmap.paste(
+                    text_bitmap, (code_bitmap.width // 2 - text_width // 2, code_bitmap.height)
+                )
+            case "right":
+                code_with_text_bitmap.paste(
+                    text_bitmap, (code_bitmap.width - text_width, code_bitmap.height)
+                )
```

Comment:
The `match` syntax is not compatible with Python 3.7 which we currently support. Instead, you could do

```suggestion
        if align == "left":
            code_with_text_bitmap.paste(text_bitmap, (0, code_bitmap.height))
        elif align == "center":
            code_with_text_bitmap.paste(
                text_bitmap, (code_bitmap.width // 2 - text_width // 2, code_bitmap.height)
            )
        elif align == "right":
            code_with_text_bitmap.paste(
                text_bitmap, (code_bitmap.width - text_width, code_bitmap.height)
            )
```